plugins/source/gcp/resources/services/networkconnectivity/locations.go

```diff
@@ -0,0 +1,52 @@
+package networkconnectivity
+
+import (
+	"context"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"google.golang.org/api/networkconnectivity/v1"
+
+	"github.com/cloudquery/cloudquery/plugins/source/gcp/client"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+)
+
+func Locations() *schema.Table {
+	return &schema.Table{
+		Name:        "gcp_networkconnectivity_locations",
+		Description: `https://cloud.google.com/network-connectivity/docs/reference/networkconnectivity/rest/v1/projects.locations/list`,
+		Resolver:    fetchLocations,
+		Multiplex:   client.ProjectMultiplexEnabledServices("networkconnectivity.googleapis.com"),
+		Transform:   client.TransformWithStruct(&networkconnectivity.Location{}, transformers.WithPrimaryKeys("Name")),
+		Columns: []schema.Column{
+			{
+				Name:       "project_id",
+				Type:       arrow.BinaryTypes.String,
+				Resolver:   client.ResolveProject,
+				PrimaryKey: true,
+			},
+		},
+		Relations: []*schema.Table{
+			InternalRanges(),
+		},
+	}
+}
+
+func fetchLocations(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
```

Comment:
This requires a mock test, see examples in other resources (kms, for instance).

Reply:
Seems strange that proto lib doesn't expose `Internal-Range` and `Location` .pb file. but our mock requires that.

Reply:
Locations are from more common place, see [kms locations test](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/gcp/resources/services/kms/locations_mock_test.go)

Additionally, you might want to use the REST test helper instead of GRPC one, see https://github.com/cloudquery/cloudquery/blob/main/plugins/source/gcp/client/testing.go